sdk/search/Azure.Search/src/Models/SearchResult{T}.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Search.Models
+{
+    // Hide the untyped SearchResult
+    [CodeGenSchema("SearchResult")]
+    internal partial class SearchResult { }
+
+    /// <summary>
+    /// Contains a document found by a search query, plus associated metadata.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The .NET type that maps to the index schema. Instances of this type can
+    /// be retrieved as documents from the index.
+    /// </typeparam>
+    [JsonConverter(typeof(SearchResultConverterFactory))]
+    public class SearchResult<T>
+    {
+        /// <summary>
+        /// The relevance score of the document compared to other documents
+        /// returned by the query.
+        /// </summary>
+        public double? Score { get; internal set; }
+
+        /// <summary>
+        /// Text fragments from the document that indicate the matching search
+        /// terms, organized by each applicable field; null if hit highlighting
+        /// was not enabled for the query.
+        /// </summary>
+        public IDictionary<string, IList<string>> Highlights { get; internal set; }
```

Comment:
Should this be mutable?

Reply:
I'm tracking the generator in terms of these sort of things so I'll make mine immutable once they do.  I don't want the models to be half one way or the other.